ReSharper.FSharp/src/FSharp/FSharp.Psi/src/Impl/FSharpImplUtil.cs

```diff
@@ -171,6 +171,15 @@ public static string GetCompiledName([CanBeNull] this IIdentifier identifier)
     public static string GetSourceName([CanBeNull] this IIdentifier identifier) =>
       identifier?.Name ?? SharedImplUtil.MISSING_DECLARATION_NAME;
 
+    [NotNull]
+    public static string GetTypeParameterOrSourceName([CanBeNull] this IIdentifier identifier)
```

Comment:
This whole method looks like an implementation detail of `ITypeParameterId`. I think we should adjust the type.

Reply:
removed